eth/downloader/downloader.go

```diff
@@ -1008,18 +1008,17 @@ func (d *Downloader) findAncestor(p *peer) (uint64, error) {
 				}
 			}
 
+		case <-timeout:
+			glog.V(logger.Debug).Infof("%v: head header timeout", p)
+			return 0, errTimeout
+
 		case <-d.bodyCh:
```

Comment:
Same as above, ancestor lookup is ran before the fetchers start, so it's safe to deplete all data delivery channels.